fv3core/stencils/delnflux.py

```diff
@@ -1,94 +1,404 @@
+                (local_js - 0 - 1) : (local_je + 0 + 2),
+            ]
+        ):
+            d2 = d2_high_order(fx, fy, rarea)
+        d2 = corners.copy_corners_x(d2)
+        with horizontal(
+            region[
+                (local_is - 0) : (local_ie + 0 + 2), (local_js - 0) : (local_je + 0 + 1)
+            ]
+        ):
+            fx = fx_calculation(d2, del6_v, 4)
+        d2 = corners.copy_corners_y(d2)
+        with horizontal(
+            region[
+                (local_is - 0) : (local_ie + 0 + 1), (local_js - 0) : (local_je + 0 + 2)
+            ]
+        ):
+            fy = fy_calculation(d2, del6_u, 4)
+
+
+@gtstencil
+def higher_order_compute_unroll2(
+    fx: FloatField,
+    fy: FloatField,
+    rarea: FloatField,
+    d2: FloatField,
+    del6_u: FloatField,
+    del6_v: FloatField,
+):
+    from __externals__ import local_ie, local_is, local_je, local_js
+
+    with computation(PARALLEL), interval(...):
+
+        d2 = corners.copy_corners_x(d2)
+        with horizontal(
+            region[
+                (local_is - 2) : (local_ie + 2 + 2), (local_js - 2) : (local_je + 2 + 1)
+            ]
+        ):
+            fx = fx_calculation(d2, del6_v, 1)
+        d2 = corners.copy_corners_y(d2)
+        with horizontal(
+            region[
+                (local_is - 2) : (local_ie + 2 + 1), (local_js - 2) : (local_je + 2 + 2)
+            ]
+        ):
+            fy = fy_calculation(d2, del6_u, 1)
+
+        with horizontal(
+            region[
+                (local_is - 1 - 1) : (local_ie + 1 + 2),
+                (local_js - 1 - 1) : (local_je + 1 + 2),
+            ]
+        ):
+            d2 = d2_high_order(fx, fy, rarea)
+        d2 = corners.copy_corners_x(d2)
+        with horizontal(
+            region[
+                (local_is - 1) : (local_ie + 1 + 2), (local_js - 1) : (local_je + 1 + 1)
+            ]
+        ):
+            fx = fx_calculation(d2, del6_v, 2)
+        d2 = corners.copy_corners_y(d2)
+        with horizontal(
+            region[
+                (local_is - 1) : (local_ie + 1 + 1), (local_js - 1) : (local_je + 1 + 2)
+            ]
+        ):
+            fy = fy_calculation(d2, del6_u, 2)
+
+        with horizontal(
+            region[
+                (local_is - 0 - 1) : (local_ie + 0 + 2),
+                (local_js - 0 - 1) : (local_je + 0 + 2),
+            ]
+        ):
+            d2 = d2_high_order(fx, fy, rarea)
+        d2 = corners.copy_corners_x(d2)
+        with horizontal(
+            region[
+                (local_is - 0) : (local_ie + 0 + 2), (local_js - 0) : (local_je + 0 + 1)
+            ]
+        ):
+            fx = fx_calculation(d2, del6_v, 3)
+        d2 = corners.copy_corners_y(d2)
+        with horizontal(
+            region[
+                (local_is - 0) : (local_ie + 0 + 1), (local_js - 0) : (local_je + 0 + 2)
+            ]
+        ):
+            fy = fy_calculation(d2, del6_u, 3)
```

Comment:
It would be helpful for me to see the unrolled versions with functions, so it is easier to see what is repeated and altered.

Reply:
We can abstract most of the unrolled loop calculations, but the first one explicitly depends on nord, so it would be a separate function for each unrolled loop. I've tried to clean it up in the latest commit